improved computation of the contact wrench cone for efficiency

rjgriffin42 opened the pull request.

james-p-foster committed:
Add method for adding cost item to cost sum via shared pointer

james-p-foster committed:
Fixed missing symbol

rjgriffin42 committed:
improved computation of the contact wrench cone for efficiency

rjgriffin42 commented on include/crocoddyl/multibody/wrench-cone.hxx:
```diff
@@ -142,9 +142,12 @@ void WrenchConeTpl<Scalar>::update() {
   Scalar theta = Scalar(2) * M_PI / static_cast<Scalar>(nf_);
   if (inner_appr_) {
-    mu *= cos(theta / Scalar(2.));
+    mu *= cos(theta * Scalar(0.5));
```
There's a thing about multiplication operations being more efficient than division operations depending on the processor. The other is never the case.

cmastalli commented:
Hi @rjgriffin42 ! I suggest not duplicating PRs, as the IHMC fork will quickly diverge. Many changes will come in Crocoddyl. Therefore, this should be avoided.

rjgriffin42 commented:
Carlos, I'm not sure what you're suggesting. I didn't set up a PR on the base of crocoddyl, so it's not a duplicated PR. Are you suggesting that I should, instead, have the PR go to the base of crocoddyl? Because I think there's a number of things on the IHMC fork that should be incorporated upstream, too.

cmastalli commented:
> Carlos, I'm not sure what you're suggesting. I didn't set up a PR on the base of crocoddyl, so it's not a duplicated PR. Are you suggesting that I should, instead, have the PR go to the base of crocoddyl? Because I think there's a number of things on the IHMC fork that should be incorporated upstream, too.

Hi @rjgriffin42!

1. You have indeed created a duplicate PR in the root repo of Crocoddyl (see https://github.com/loco-3d/crocoddyl/pull/1343)
2. I suggest closing this PR and not pushing anything here anymore. See my comments above of why not doing this.
3. I have reviewed this PR in the base repo. 

For instance, your current PR is based on an old version of Crocoddyl based on `boost::shared_ptr`. However, we have recently migrated to `std::shared_ptr`. This is not yet released, but we decided to make a major release soon with it as it breaks API.

Moreover, we are making huge changes in Crocoddyl (inside our private repo for now):
  1. Adding dynamics, residuals, and solvers for optimal estimation.
  2. Parametrizing our MPC solver (e.g., for time-opt problems)
  3. Adding casting methods for code-generation in Python and running solvers with `floats` or `brain floats`.
  4. Developing two advanced constrained solvers in parallel.

My PhD students and I are working hard on these topics. Many of these things are working now, but we're still making improvements, such as refining our current API for better efficiency and user experience.

In short, if we keep pushing code to this fork, we'll have a headache later when integrating the new version of Crocoddyl.

If you want to change things in Crocoddyl, let's discuss coordinating efforts while avoiding headaches. Such changes can be done inside our private repo or in the public one, depending on their nature.

cmastalli committed:
[action] Added action codegen profiler macros

cmastalli committed:
[ci] Updated conda job for helping debugging

cmastalli committed:
Missed header in action codegen

cmastalli committed:
[codegen] Simplified action codegen API thanks to our casting method

cmastalli committed:
[codegen] Initialized unique pointers

cmastalli committed:
[codegen] Implemented all virtual functions in action codegen

cmastalli committed:
[python] Create an enum for casting between scalars

cmastalli committed:
[bench] Updated bench with API change in action codegen

cmastalli committed:
[python] Allowed casting of NAN to scalars different floating-point types

cmastalli committed:
[core] Added missed Scalar(.)

cmastalli committed:
[cleanup] Define Scalar in Python wrap classes

cmastalli committed:
[core] Introduced ScalarSelector to select the floating-base point scalar associated to Scalar + used in case we don't need to use codegen

cmastalli committed:
[core] Defined isfinite to handle CppAD types

cmastalli committed:
[cleanup] Moved math.hpp to fwd.hpp files

cmastalli committed:
[core] Fixed casting in wrench's scalars

cmastalli committed:
[math] Defined pow and sqrt dependending on scalar type

cmastalli committed:
[cleanup] Moved exception.hpp to fwd.hpp files

cmastalli committed:
[codegen] Don't cast to CppAD for action codegen class

cmastalli committed:
[cleanup] Remove using std::isfinite

cmastalli committed:
[core] Overload Eigen::pow/sqrt for CppAD-compatible types

cmastalli committed:
[core] Enabled CppGG casting (this requires Pinocchio fixes)

cmastalli committed:
Missed this

cmastalli committed:
[core] Fixed constructor needed for CppAD casting

cmastalli committed:
[core] Resolved scalar casting ambiguity with handling CppAD types

cmastalli committed:
[python] Return self when casting to same type

cmastalli committed:
[python] Introduced CppAD python bindings

cmastalli committed:
[action] Fixed MatrixXs declaration

cmastalli committed:
[codegen] Fixed quasiStatic in action codegen

cmastalli committed:
[codegen] Added non-CG calc and calcDiff for terminal nodes

cmastalli committed:
[python] Created Python bindings for codegen action

cmastalli committed:
[cleanup] Reused CROCODDYL_PYTHON_FLOATINGPOINT_SCALARS when CROCODDYL_WITH_CODEGEN=False

cmastalli committed:
[cmake] Enabled ccache if we found it

cmastalli committed:
[core] Supported CppAD types in pseudoInverse

cmastalli committed:
[cleanup] Removed useless MatrixXs constructor

cmastalli committed:
[codegen] Fixed copy constructor in action codegen

cmastalli committed:
[python] Binds compiler type used in action codegen

cmastalli committed:
[core] Added Eigen casting from and to CppD types

cmastalli committed:
[codegen] Fully implemented the casting approach in codegen action

cmastalli committed:
[actuation] Supported codegen when defining tau_set in actuation thruster

cmastalli committed:
[math] Added fabs method to support codegen

cmastalli committed:
[python] Exposed action in codegen data

cmastalli committed:
[codegen] Supported constraints in codegen action

cmastalli committed:
[core] Support c++11 like pow by allowing passing an int exponential

cmastalli committed:
[cleanup] Reorganizing math.h file

cmastalli committed:
[cleanup] Removed math.h dependency by introducing pi constant

cmastalli committed:
[core] Defined isfinite

cmastalli committed:
[cleanup] Removed useless math operations

cmastalli committed:
[cleanup] Constructing Scalar(1.) rather than casting it

cmastalli committed:
[state] Used max for defined root_joint position limits + fixed compution of root joint DoF

cmastalli committed:
[math] Defined isfinite for CppAD types

cmastalli committed:
[cleanup] Reorganized header files to attend solving issues with some compilers

cmastalli committed:
[action] Supported codegen in contact-invdyn fix

cmastalli committed:
[wip] Tested if this fixes job compilation error

cmastalli committed:
[cleanup] Removed template-id in data-contact constructor (not allowed in C++20)

cmastalli committed:
[unittest] Added crocoddyl namespace in test_actuation's pseudoInverse

cmastalli committed:
[unittest] Used crocoddyl namespace in test_friction

cmastalli committed:
[logs] Updated logs after rebasing

cmastalli committed:
[bugfix] Fixed (diff)-LQR's set_LQR function

cmastalli committed:
[logs] Updated logs after rebasing

cmastalli committed:
[changelog] Update CHANGELOG.md

cmastalli committed:
[problem] Casted the nthreads in shooting problem

cmastalli committed:
[macro] Removed duplicated macros + added missed one

Commit:
Remove absolute path for boost library

cmastalli committed:
[action] Extend LQR's check for semi-postive matrices

cmastalli committed:
Update CHANGELOG.md

Commit:
Add install version.

This will install libcrocoddyl.so.2.2.0 as a binary and libcrocoddyl.so
as a symlink, instead of just libcrocoddyl.so as a binary.

cmastalli committed:
[core] Introduced ScaleNumerics for adapting numbers from doubles to floats

cmastalli committed:
[core] Adapted scale in numerics

cmastalli committed:
[ci] Limited number of building jobs in Nix

cmastalli committed:
[core] Check psd for floating-point scalars only (Eigen doesn't support CppAD types)

cmastalli committed:
[core] Apply SIMD in shooting problem in floating-point numbers only

cmastalli committed:
[pinocchio] Added missed casting in Pinocchio

cmastalli committed:
[pinocchio] Added missed casting in Pinocchio + included codegen in fwd

cmastalli committed:
[pinocchio] Disabled Debug mode unit tests that triggers quaternion asserts with floats (needed to handle this properly in Pinocchio)

cmastalli committed:
[ci] Reverted Nix CI changes

cmastalli committed:
Revert "[cmake] Enabled ccache if we found it"

This reverts commit d1a121209e783d9ec0b9c7d73cdd3e41851322d9.

cmastalli committed:
Merge pull request #1346 from cmastalli/topic/casting

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.9.10 → v0.11.0](https://github.com/astral-sh/ruff-pre-commit/compare/v0.9.10...v0.11.0)
- [github.com/pre-commit/mirrors-clang-format: v19.1.7 → v20.1.0](https://github.com/pre-commit/mirrors-clang-format/compare/v19.1.7...v20.1.0)

cmastalli committed:
[pre-commit] Fixed pre-commit by using tuple comprehension

nim65s committed:
sync submodule

nim65s committed:
pre-commit autoupdate

nim65s committed:
nix: update flake

nim65s committed:
CI: drop ROS 1

nim65s committed:
CI: update ros distros

nim65s committed:
ROS CI: pin

nim65s committed:
require pinocchio >= 3.4.0

nim65s committed:
update example-robot-data

fix https://github.com/Gepetto/example-robot-data/issues/270

nim65s committed:
fix changelog

cmastalli committed:
Merge pull request #1361 from loco-3d/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

nim65s committed:
release: Update package.xml version to 3.0.0

nim65s committed:
release: Update CHANGELOG.md for 3.0.0

nim65s committed:
Merge branch 'master' into devel

Sergim96 committed:
[bindings]: constraint dimension getter as virtual

Sergim96 committed:
[bindings]: ddeprecate constraint dimension setter

Sergim96 committed:
add override keyword

cmastalli committed:
[changelog] Update CHANGELOG.md

cmastalli committed:
[ci] Reverted to v3.0.0's ROS CI config

cmastalli committed:
Merge pull request #1362 from Sergim96/topic/virtual_binding_constraint_dimension

nim65s committed:
nix flake update

nim65s committed:
pre-commit autoupdate

nim65s committed:
fix .git-blame-ignore-revs for github UI

nim65s committed:
release: Update package.xml version to 3.0.1

nim65s committed:
release: Update CHANGELOG.md for 3.0.1

nim65s committed:
Merge branch 'master' into devel

cmastalli committed:
Merge pull request #1364 from cmastalli/topic/ros-ci

Fixing ROS CI in devel branch

cmastalli committed:
[cost] Used std::shared in addCost

cmastalli committed:
[core] Introduced template instantiation

cmastalli committed:
[bugfix] Fixed vector dim in cones

cmastalli committed:
[bugfix] Fixed diff-LQR's set_LQR tolerance

cmastalli committed:
[changelog] Updated changelog

cmastalli committed:
[unittest] Improved contact numdiff tolerance for floats

cmastalli committed:
[unittest] Fixed tolerance of da0_dx in floats

cmastalli committed:
[ci] Don't specify the python version for Conda jobs

cmastalli committed:
[ci] Removed eigen version dependency in Conda jobs

cmastalli committed:
[cmake] Imposing Pinocchio v3.1.0

MaximilienNaveau committed:
fix cmake warning during python_build

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

MaximilienNaveau committed:
Update changelog

cmastalli committed:
[core] Added explicit template instantiation of used-multiple-time numdiff classes

cmastalli committed:
[deprecate] Removed ShootingProblem's get_nu_max function (previously deprecated)

cmastalli committed:
Merge pull request #1369 from MaximilienNaveau/bugfix/mnaveau/python-build

Bugfix: Fix CMake error message during `python_build`

cmastalli committed:
Merge branch 'devel' into topic/wip-eti

cmastalli committed:
[core] Used CROCODDYL_DECLARE_EXTERN_TEMPLATE_CLASS in missed classes/structures

cmastalli committed:
[cleanup] Apply properly Scalar in action numdiff

cmastalli committed:
[cleanup] Used "using Base::torqueTransform"

cmastalli committed:
[core] Fixed actuation-base.cpp.in

cmastalli committed:
[cleanup] Renamed macro to CROCODDYL_DERIVED_FLOATINGPOINT_CAST

cmastalli committed:
[core] Created mechanism to handle instantiation with floating-point only

cmastalli committed:
[cmake] Turn off boost-related cmake warning message

cmastalli committed:
[core] Define action codegen when ifdef CROCODDYL_WITH_CODEGEN

cmastalli committed:
Merge pull request #1367 from cmastalli/topic/wip-eti

Introducing explicit template instantiation

cmastalli committed:
[core] Introduced pch for core and  mutibody files + cleaned up header inclusions

cmastalli committed:
[bugfix] Fixed checkPSD in LQR actions

cmastalli committed:
[codegen] Codegenerated terminal calc/calcDiff functions

cmastalli committed:
[codegen] Added virtual keyword + improved documentation

cmastalli committed:
[codegen] Codegenerated quasiStatic + improved action codegen documentation

cmastalli committed:
[python] Exposed missed action codegen objects

cmastalli committed:
[codegen] Introduced autodiff in codegen action

cmastalli committed:
[codegen] Simplify codegen action code

cmastalli committed:
[core] Flushing callback and profiler prints

cmastalli committed:
[codegen] Enabled multithreading in codegen with a Python code

cmastalli committed:
[cleanup] Renamed nP as np + documented it

cmastalli committed:
[deprecate] Renamed codegen/action-base.hpp to codege/action.hpp + deprecated previous header

cmastalli committed:
[cleanup] Renamed action-base.cpp_fptype.in to action.cpp_fptype.in

cmastalli committed:
[cleanup] Removed useless header inclusion in diff-lqr.hxx

cmastalli committed:
[codegen] Splitted action codegen into hpp and hxx files

cmastalli committed:
[cleanup] Removed crocoddyl namespace in codegen-action typedef

cmastalli committed:
[codegen] Renamed set_parameters to update_p

cmastalli committed:
[codegen] Added action codegen signatures in Python

cmastalli committed:
[codegen] Enabled passing environments in Python

cmastalli committed:
[codegen] Avoid allocating data of Hessian selectors

cmastalli committed:
[codegen] Fixed update_p for terminal nodes

cmastalli committed:
[codegen] Checked p size in update_p

cmastalli committed:
[notebooks] Improved and structuring notebooks

cmastalli committed:
[notebooks] Created a folder for notebooks + updated readme

cmastalli committed:
[cmake] Removed pycppad dependency

cmastalli committed:
[ci] Replaced Conda's boost dependency as suggedted by @traversaro

cmastalli committed:
[codegen] Included forward declaration in pinocchio_cast.hpp

cmastalli committed:
[cmake] Cleaned up the cmake file

cmastalli committed:
[changelog] Updated

cmastalli committed:
[bugfix] Fixed compilation errors in Conda's CI jobs

cmastalli committed:
[cmake] Used CONFIGURE_DEPENDS to ensure that CMake re-scans the directory

cmastalli committed:
[codegen] Removed useless action data

cmastalli committed:
[codegen] Check existence and load a library based on its name

cmastalli committed:
[codegen] Created mechanism to allows us pre-compiled loading libraries

cmastalli committed:
[codegen] Optimizing setter functions in codegen data

cmastalli committed:
[codegen] Avoided define wCostHess each time

cmastalli committed:
Merge pull request #1370 from cmastalli/topic/cppad

cmastalli committed:
[python] Enabled t setup the cast visitor for floating-point scalars only

cmastalli committed:
[python] Introduced bindings for multi-scalar core-numdiff classes

cmastalli committed:
[python] Created independent Python libraries per each scalar

cmastalli committed:
[python] Compiled solvers for float64 only

cmastalli committed:
[codegen] Handled floating-point scalar only in action codegen

cmastalli committed:
[macro] Removed non-needed Python macros

cmastalli committed:
[notebook] Added details on `BUILD_WITH_CODEGEN_SUPPORT` in codegen notebook

cmastalli committed:
[cleanup] Do not duplicate the typedef of scalars

cmastalli committed:
[unittest] Updated test_states.py

cmastalli committed:
[nix] Added crocoddyl.cmake to the build environment

cmastalli committed:
[changelog] Updated changelog

cmastalli committed:
[ci] Removed cppad dependency in conda env

cmastalli committed:
[python] Fixed Python stubs generation

cmastalli committed:
[pinocchio] Hide casting for Pinocchio >v3.5.0

cmastalli committed:
[cmake] Cleaned up cmake scalar order

cmastalli committed:
[pinocchio] Removed code supporting Pinocchio v3.0.0<

cmastalli committed:
Merge pull request #1372 from cmastalli/topic/python-scalar-libs

Created independent Python modules associated to each scalar

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.11.2 → v0.11.4](https://github.com/astral-sh/ruff-pre-commit/compare/v0.11.2...v0.11.4)

cmastalli committed:
Merge pull request #1373 from loco-3d/pre-commit-ci-update-config

cmastalli committed:
[python] Needed to add actions' getter bindings after making them virtual

cmastalli committed:
[cleanup] Fixed grammar in documentation

cmastalli committed:
[unittest] Enforce to run action getters in Python

cmastalli committed:
[core] Compile floating-point for the shooting problem only.

cmastalli committed:
[changelog] Updated changelog

cmastalli committed:
[cmake] Don't generate action codegen cpp if not built with codegen

cmastalli committed:
[pinocchio] No need to register pinocchio model in >v3.0.0

cmastalli committed:
Merge pull request #1374 from cmastalli/topic/fix-getter-bindings

cmastalli committed:
[ci] Enable ubuntu-latest in conda + remove debug symbols

cmastalli committed:
[cmake] Applies CMP0074 policy for >3.12

[cmake] Make CMP0074 unconditional

[cmake] Set CMP0074 in the beginning

cmastalli committed:
Merge pull request #1376 from cmastalli/topic/ci-conda

Conda CI improvements

cmastalli committed:
[cmake] Improved cmake printing

cmastalli committed:
[core] Added pycppad as dependency when building with codegen support

cmastalli committed:
[ci] Defined pycppad as dependency of Conda jobs

cmastalli committed:
[unittest] Tested notebooks

cmastalli committed:
[cmake] Removed "s" in cmake custom commands

cmastalli committed:
[readme] Updated publications + removed steering comittee section

cmastalli committed:
[logs] Updated logs

cmastalli committed:
[cmake] Removed PKG_CONFIG_REQUIRES in pycppad project dependency + updated gepetto webpage

nim65s committed:
flake: add notebooks

nim65s committed:
flake: need nbformat for tests

nim65s committed:
flake: need nbconvert for tests

cmastalli committed:
[cmake] Fixed the python path for running notebook tests

cmastalli committed:
Merge branch 'topic/pycppad-and-notebooks' of github.com:cmastalli/crocoddyl into topic/pycppad-and-notebooks

github-actions[bot] committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'flake-parts':
    'github:hercules-ci/flake-parts/f4330d22f1c5d2ba72d3d22df5597d123fdb60a9?narHash=sha256-%2Bu2UunDA4Cl5Fci3m7S643HzKmIDAe%2BfiXrLqYsR2fs%3D' (2025-03-07)
  → 'github:hercules-ci/flake-parts/c621e8422220273271f52058f618c94e405bb0f5?narHash=sha256-hIshGgKZCgWh6AYJpJmRgFdR3WUbkY04o82X05xqQiY%3D' (2025-04-01)
• Updated input 'flake-parts/nixpkgs-lib':
    'github:nix-community/nixpkgs.lib/147dee35aab2193b174e4c0868bd80ead5ce755c?narHash=sha256-oiwv/ZK/2FhGxrCkQkB83i7GnWXPPLzoqFHpDD3uYpk%3D' (2025-03-02)
  → 'github:nix-community/nixpkgs.lib/e4822aea2a6d1cdd36653c134cacfd64c97ff4fa?narHash=sha256-b1EdN3cULCqtorQ4QeWgLMrd5ZGOjLSLemfa00heasc%3D' (2025-03-30)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/a84ebe20c6bc2ecbcfb000a50776219f48d134cc?narHash=sha256-mNqIplmEohk5jRkqYqG19GA8MbQ/D4gQSK0Mu4LvfRQ%3D' (2025-03-19)
  → 'github:NixOS/nixpkgs/f675531bc7e6657c10a18b565cfebd8aa9e24c14?narHash=sha256-gbl9hE39nQRpZaLjhWKmEu5ejtQsgI5TWYrIVVJn30U%3D' (2025-04-09)

cmastalli committed:
[flake] Needed ipykernel for tests

cmastalli committed:
[ci] Defined minimal Pinocchio version

cmastalli committed:
[ci] Added Conda dependencies for notebooks tests

cmastalli committed:
[ci] Added matplotlib for Nix tests

cmastalli committed:
Revert "[pinocchio] No need to register pinocchio model in >v3.0.0"

This reverts commit c5e92d9d5714fa1a98153dcffc67226c825ae8b6.

cmastalli committed:
[core] Do not register Pinocchio model for <v3.4.0

cmastalli committed:
[ci] Added ffmpeg as Nix test dependency

cmastalli committed:
[ci] Added ffmpeg as conda test dependency

cmastalli committed:
[ci] Updated rosdep to handle notebook tests

cmastalli committed:
[ci] Added ipykernel in conda tests

cmastalli committed:
[ci] Added matplotlib as conda dependency

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.11.4 → v0.11.5](https://github.com/astral-sh/ruff-pre-commit/compare/v0.11.4...v0.11.5)

cmastalli committed:
[ci] Fixed coda_support flat in conda jobs

cmastalli committed:
[core] Replace codegen header

cmastalli committed:
Merge pull request #1379 from loco-3d/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

cmastalli committed:
[ci] Relying on cxx_compiler for Conda jobs

cmastalli committed:
[core] Simplify CppAD headers

cmastalli committed:
[core] Removed inclusion of conversions.hpp

cmastalli committed:
[core] Included ppad/cppad.hpp instead of cppad/cg/cppadcg.hpp

cmastalli committed:
Merge pull request #1377 from cmastalli/topic/pycppad-and-notebooks

Defined pycppad as codegen dependency + unittest notebooks + update readme

cmastalli committed:
Merge pull request #1378 from loco-3d/update_flake_lock_action

cmastalli committed:
Merge remote-tracking branch 'origin/master' into devel

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.11.5 → v0.11.6](https://github.com/astral-sh/ruff-pre-commit/compare/v0.11.5...v0.11.6)

dependabot[bot] committed:
Bump ros-industrial/industrial_ci

Bumps [ros-industrial/industrial_ci](https://github.com/ros-industrial/industrial_ci) from 3e67ec54d63496e076267392148a26229740befc to cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8.
- [Commits](https://github.com/ros-industrial/industrial_ci/compare/3e67ec54d63496e076267392148a26229740befc...cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8)

---
updated-dependencies:
- dependency-name: ros-industrial/industrial_ci
  dependency-version: cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

cmastalli committed:
Merge pull request #1381 from loco-3d/dependabot/github_actions/devel/ros-industrial/industrial_ci-cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8

cmastalli committed:
Merge pull request #1380 from loco-3d/pre-commit-ci-update-config

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.11.6 → v0.11.7](https://github.com/astral-sh/ruff-pre-commit/compare/v0.11.6...v0.11.7)

cmastalli committed:
Merge pull request #1383 from loco-3d/pre-commit-ci-update-config

dependabot[bot] committed:
Bump ros-industrial/industrial_ci

Bumps [ros-industrial/industrial_ci](https://github.com/ros-industrial/industrial_ci) from cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8 to 16472786c1ffc53596007da4318db98d1f677933.
- [Commits](https://github.com/ros-industrial/industrial_ci/compare/cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8...16472786c1ffc53596007da4318db98d1f677933)

---
updated-dependencies:
- dependency-name: ros-industrial/industrial_ci
  dependency-version: 16472786c1ffc53596007da4318db98d1f677933
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

cmastalli committed:
Merge pull request #1384 from loco-3d/dependabot/github_actions/devel/ros-industrial/industrial_ci-16472786c1ffc53596007da4318db98d1f677933

Bump ros-industrial/industrial_ci from cfe5c929f7dbb0374c01bc26821ec8eb72f6d4e8 to 16472786c1ffc53596007da4318db98d1f677933

rjgriffin42 committed:
did some more efficiency improvements, and updated the changelog

rjgriffin42 committed:
added a test setter test

cmastalli committed:
[cone] Used the block reference in wrench cone + fixed A computation

cmastalli committed:
[python] Fixed compilation issues introduced with the new addCost bindings

cmastalli committed:
[cone] Propagated wrench-computation improvement to the friction cone
